src/library/udp_proxy.cpp

```diff
@@ -81,7 +82,8 @@ Error ProxyEndpoint::Send(const ByteArray &aRequest, MessageSubType aSubType)
 
     VerifyOrExit(mBrClient.IsConnected(), error = ERROR_INVALID_STATE("not connected to the border agent"));
 
-    utils::Encode<uint16_t>(udpPayload, mBrClient.GetDtlsSession().GetLocalPort());
+    sockPort = (mSockPort == 0) ? mBrClient.GetDtlsSession().GetLocalPort() : GetSockPort();
```

Comment:
The `mSockPort` will be always not `0` once it received a `*.ans` from an end device, because there is no code to set `mSockPort` back to `0` explicitly.

It's probably just simpler to use the const int `kDefaultMmPort` for now? Since this is the only port the commissioner should be bind to.